### DISCUSSION TITLE
Unattended install

### DISCUSSION THREAD

**Discussion by giorgiga** [Homebrew/discussions#2728] [Linux]
From the docs it seems untarring `https://github.com/Homebrew/brew/tarball/master` to `/home/linuxbrew/.linuxbrew` is a valid way to install homebrew, however that results in a different setup than running the interactive installer (the installer puts homebrew in `/home/linuxbrew/.linuxbrew/Homebrew` and then symlinks - eg. - `.linuxbrew/bin/brew` to `.linuxbrew/Homebrew/bin/brew`).

I didn't test this thoroughly, but I'm confident it works either way.

I was however wondering... Why does the installer do a seemingly more complicated setup than just uncompressing a tarball? Would there be any advantages if I replicated that rather than just untarring?

Also, the installer downloads via git clone resulting in a directory `/home/linuxbrew/.linuxbrew/Homebrew/.git` - does that serve any purpose?

**Comment by SMillerDev** [comment]
> I was however wondering... Why does the installer do a seemingly more complicated setup than just uncompressing a tarball? Would there be any advantages if I replicated that rather than just untarring?

See https://docs.brew.sh/Installation

> Also, the installer downloads via git clone resulting in a directory /home/linuxbrew/.linuxbrew/Homebrew/.git - does that serve any purpose?

Brew repositories are git repos to keep track of changes and versioning. It'll need this to work.

**Reply by giorgiga** [reply]
I did look at the doc (the one you mention and also [the linux-specific doc](https://docs.brew.sh/Homebrew-on-Linux)), but it doesn't seem to explain why the installed does what it does (and - consequently - whether doing the same is worth anything for me).

As for the .git part... I messed that up :smile: 
Turns out one gets 2 git repos either way inside the `.linuxbrew` directory: running the installer creates both, but if one installs via untar untar one is created when running `brew update` for the first time and a second (which I missed before) when one installs some package.
